backend/dal/Helpers/Extensions/BuildingExtensions.cs

```diff
@@ -24,11 +24,11 @@ public static class BuildingExtensions
         /// <exception type="DbUpdateException">The name within a parcel should be unique.</exception>
         public static void ThrowIfNotUnique(this PimsContext context, Entity.Parcel parcel, Entity.Building building)
         {
-            if(parcel == null)
+            if(parcel == null || String.IsNullOrEmpty(building.Name))
             {
                 return;
             }
-            var parcelBuildings = context.Parcels.Where(p => p.Id == parcel.Id).SelectMany(p => p.Buildings.Select(b => b.Building.Name)).Distinct().ToArray();
+            var parcelBuildings = context.Parcels.Where(p => p.Id == parcel.Id).SelectMany(p => p.Buildings).Where(b => b.BuildingId != building.Id).Select(b => b.Building.Name).Distinct().ToArray();
```

Comment:
previously updating a building without changing its name would trigger a duplicate name exception, this ensures that duplicates are checked for all of the OTHER buildings on the parcel during an update, ignoring the building being updated.